rotkehlchen/chain/avalanche/manager.py

```diff
@@ -0,0 +1,229 @@
+        self.w3 = Web3(
+            HTTPProvider(
+                endpoint_uri=avaxrpc_endpoint,
+                request_kwargs={'timeout': self.eth_rpc_timeout},
+            ),
+        )
+        self.covalent = covalent
+        self.msg_aggregator = msg_aggregator
+
+    def connected_to_any_web3(self) -> bool:
+        return self.w3.isConnected()
+
+    def get_latest_block_number(self) -> int:
+        return self.w3.eth.block_number
+
+    def get_avax_balance(self, account: ChecksumEthAddress) -> FVal:
+        """Gets the balance of the given account in AVAX
+
+        May raise:
+        - RemoteError if Covalent is used and there is a problem querying it or
+        parsing its response
+        """
+        result = self.covalent.get_token_balances_address(account)
+        if result is False:
+            balance = from_wei(FVal(self.w3.eth.get_balance(account)))
+        else:
+            def filter_avax(value: Dict) -> Any:
+                return value['contract_address'] == '0x9debca6ea3af87bf422cea9ac955618ceb56efb4'
+
+            avax_coin = list(filter(filter_avax, result))
+            balance = from_wei(FVal(avax_coin[0]['balance']))
+        return FVal(balance)
+
+    def get_multiavax_balance(
+            self,
+            accounts: List[ChecksumEthAddress],
+    ) -> Dict[ChecksumEthAddress, FVal]:
+        """Returns a dict with keys being accounts and balances in AVAX
+
+        May raise:
+        - RemoteError if an external service such as Covalent is queried and
+          there is a problem with its query.
+        """
+        balances = {}
+        for account in accounts:
+            balances[account] = self.get_avax_balance(account)
+        return balances
+
+    def get_block_by_number(self, num: int) -> Dict[str, Any]:
+        """Returns the block object corresponding to the given block number
+
+        May raise:
+        - RemoteError if an external service such as Covalent is queried and
+        there is a problem with its query.
+        """
+
+        block_data: MutableAttributeDict = MutableAttributeDict(self.w3.eth.get_block(num))  # type: ignore # pylint: disable=no-member  # noqa: E501
+        block_data['hash'] = hex_or_bytes_to_str(block_data['hash'])
+        return dict(block_data)
+
+    def get_code(self, account: ChecksumEthAddress) -> str:
+        """Gets the deployment bytecode at the given address
+
+        May raise:
+        - RemoteError if Covalent is used and there is a problem querying it or
+        parsing its response
+        """
+
+        return hex_or_bytes_to_str(self.w3.eth.getCode(account))
+
+    def get_transaction_receipt(
+            self,
+            tx_hash: str,
+    ) -> Dict[str, Any]:
+        tx_receipt = self.covalent.get_transaction_receipt(tx_hash)
+        if tx_receipt is False:
+            tx_receipt = self.w3.eth.get_transaction(tx_hash).__dict__  # type: ignore
+            return tx_receipt
```

Comment:
What exactly are you going here? I am not sure I get it.

Reply:
The query in covalent return false in case of timeout, if this can happen it runs through web3